BuildingBlocks/src/BuildingBlocks.Domain/Entity.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Backbone.BuildingBlocks.Domain.Events;
+using Backbone.BuildingBlocks.Domain.StronglyTypedIds.Records;
+
+namespace Backbone.BuildingBlocks.Domain;
+
+public abstract class Entity
+{
+    private readonly List<DomainEvent> _domainEvents = [];
+
+    public IReadOnlyList<DomainEvent> DomainEvents => _domainEvents;
+
+    public abstract StronglyTypedId Id { get; }
+
+    protected void RaiseDomainEvent(DomainEvent domainEvent)
+    {
+        _domainEvents.Add(domainEvent);
+    }
+
+    public void ClearDomainEvents()
+    {
+        _domainEvents.Clear();
+    }
+
+    public override bool Equals(object? obj)
+    {
+        if (obj is not Entity other)
+            return false;
+
+        if (ReferenceEquals(this, other))
+            return true;
+
+        if (GetRealType() != other.GetRealType())
+            return false;
+
+        return Id.Equals(other.Id);
+    }
+
+    public static bool operator ==(Entity? a, Entity? b)
+    {
+        if (a is null && b is null)
+            return true;
+
+        if (a is null || b is null)
+            return false;
+
+        return a.Equals(b);
+    }
+
+    public static bool operator !=(Entity? a, Entity? b)
+    {
+        return !(a == b);
+    }
+
+    public override int GetHashCode()
+    {
+        return (GetRealType() + Id).GetHashCode();
+    }
+
+    private Type GetRealType()
```

Comment:
You can remove this method and all usages. As far as I can remember it's kind of a special case in the repo we copied the code from.
But we definitely don't have types in the `Castle` namespace. :)